apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPool.TargetedSessionPool.cs

```diff
@@ -242,10 +250,20 @@ private Task<PooledSession> GetSessionAcquisitionTask(ModeOneofCase transactionM
                     else
                     {
                         // Otherwise, drop out of the lock to create a session from scratch "inline" (still asynchronously, but not via an in-flight request).
+                        // This means basically to try and nurse the pool back to health.
                     }
                 }
                 // Effectively the "else" block above, but outside the lock
-                return CreatePooledSessionAsync(cancellationToken);
+
+                // Just to hide the "await" here so we don't have to make this method async.
+                // Because we cannot await inside the lock, the only way to make this method
+                // async is to have it return Task<Task<PooledSession>> which is weird.
+                return Task.Run(async () =>
```

Comment:
I think we should be able to do this just by having an async method - I don't think we need Task.Run